libs/vm-modules/tests/unit/ml/ml_model_tests.cpp

```diff
@@ -1137,6 +1165,46 @@ TEST_F(VMModelTests, model_sequential_flatten_2d_in_2d_out)
   EXPECT_TRUE(constructed_shape == expected.shape());
 }
 
+TEST_F(VMModelTests, model_sequential_reshape_3d_in_2d_out)
+{
+  static char const *SRC_METRIC = R"(
+              function main() : Tensor
+                var shape = Array<UInt64>(1);
+                shape[0] = 1u64;
+                var x = Tensor(shape);
+                var str_vals = "0.5, 7.1, 9.1; 6.2, 7.1, 4.;";
+                x.fromString(str_vals);
+                x = x.unsqueeze();
+                var to_shape = Array<UInt64>(2);
+                to_shape[0] = 6u64;
+                to_shape[1] = 1u64;
+
+                var model = Model("sequential");
+                model.add("reshape", to_shape);
+                model.compile("scel", "adam");
+                var prediction = model.predict(x);
+                print(prediction.toString());
+
+                return prediction;
+              endfunction
+      )";
+
+  Variant res;
+  ASSERT_TRUE(toolkit.Compile(SRC_METRIC));
+  ASSERT_TRUE(toolkit.Run(&res, ChargeAmount{0}));
+  ASSERT_EQ(stdout.str(),
+            "0.500000000;"
+            "6.199999999;"
+            "7.099999999;"
+            "7.099999999;"
+            "9.099999999;"
+            "4.000000000;");
+  auto const tensor            = res.Get<Ptr<fetch::vm_modules::math::VMTensor>>();
+  auto const constructed_shape = tensor->shape();
+  fetch::math::Tensor<fetch::fixed_point::fp64_t> expected({6, 1});
+  EXPECT_TRUE(constructed_shape == expected.shape());
```

Comment:
test seems pretty reasonable. I would feel a bit more comfortable if there are one or two other tests that just check the shapes of tensor reshape (e.g. 1D -> 3D, 3D -> 2D, 10D->2D, and an invalid reshape option where the sizes are not equal? )

Reply:
I've added more tests. Take a look please

Reply:
thanks that's great